quodlibet/util/matcher.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2021 Joschua Gandert
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+from difflib import SequenceMatcher
+from operator import itemgetter
+from typing import Generic, TypeVar, Mapping, Callable, Any, Union, List, Sequence, \
+    Optional, Tuple
+
+
+T = TypeVar('T')
+Real = Union[int, float]
+AttributeGetter = Callable[[T], Any]
+AttributeGetterToWeight = Mapping[AttributeGetter, Real]
+
+SUPPORTED_NUMBER_TYPES = (int, float)
+
+
+class _MatchData:
```

Comment:
No type annotations here so hard to say, but is this not generic on the  values that it matches on?

Reply:
Thank you for reviewing! It is generic and I'll add the type hint.